File: 148-sort-list/sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode sortList(ListNode head) {
        ListNode temp=head;
        List<Integer> list=new ArrayList<>();
        while(temp!=null){
            list.add(temp.val);
            temp=temp.next;
        }
        Collections.sort(list);
        ListNode newNode=new ListNode(0);
        ListNode t=newNode;
        for(int i=0;i<list.size();i++){
            // ListNode tempnode=new ListNode(newNode.next);
            t.next=new ListNode(list.get(i));
            t=t.next;
        }
        // head=t;
        // return newNode;
        return newNode.next;
    }
}

File: 148-sort-list/sort-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def sortList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        temp=head
        lst=[]
        while temp:
            lst.append(temp.val)
            temp=temp.next
        
        lst.sort()
        dummy=ListNode(0)
        curr=dummy
        for x in lst:
            curr.next=ListNode(x)
            curr=curr.next

        return dummy.next
